pkg/git_provider/gitlab.go

```diff
@@ -0,0 +1,356 @@
+package git_provider
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/http"
+	"strings"
+
+	"github.com/quickube/piper/pkg/conf"
+	"github.com/quickube/piper/pkg/utils"
+
+	"github.com/xanzy/go-gitlab"
+)
+
+type GitlabClientImpl struct {
+	client *gitlab.Client
+	cfg    *conf.GlobalConfig
+}
+
+func NewGitlabClient(cfg *conf.GlobalConfig) (Client, error) {
+	ctx := context.Background()
+
+	client, err := gitlab.NewClient(cfg.GitProviderConfig.Token)
+	if err != nil {
+		log.Fatalf("Failed to create client: %v", err)
+	}
+
+	err = ValidateGitlabPermissions(ctx, client, cfg)
+	if err != nil {
+		return nil, fmt.Errorf("failed to validate permissions: %v", err)
+	}
+
+	group, resp, err := client.Groups.GetGroup(cfg.OrgName, nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get organization: %v", err)
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("failed to get organization data %s", resp.Status)
+	}
+
+	cfg.OrgID = int64(group.ID)
```

Comment:
I don't think that we need Org validation in Gitlab because it is on-premise GitProvider